### DISCUSSION TITLE
How to create parser only package

### DISCUSSION THREAD

**Discussion by brsanthu** [eclipse-langium/langium#1483] [Q&A]
We are working on a DSL that is meant to be used only in the server side hence we don't need all editor related aspects, including any integration with monaco. Once that package is created, we would publish to npm for other server services to pull from.

Can you please point me to some guide or share some tips on how to go about it?

**Comment by msujew** [comment]
Hey @brsanthu,

so there are two points to this question:

1. Creating a non-LSP service can be accomplished by using the `LangiumCoreServices` from the `langium` import instead of the `LangiumServices` from `langium/lsp`. This way you don't even need to import the LSP services. This cuts down on initialization time and bundle size (in case you care about that).
2. Using Langium from a server is fairly straightforward. See [here](https://github.com/TypeFox/langium-lox/blob/main/src/interpreter/runner.ts#L84-L96) which is the code that loads and disposes documents for the lox-language interpreter. While the language server takes care of document handling by itself, in a server you need to do that yourself. But it's not too complicated.

**Reply by brsanthu** [reply]
Hi @msujew,  tried to refactor to remove references to `langium/lsp` and import `LangiumCoreServices` but in vain. Some of the needed imports like `createDefaultSharedModule` and `createDefaultModule` are missing in `langium`

If you can help with working snippet, I can create a showcase repo for parser-only usage.

**Reply by msujew** [reply]
Instead of `createDefaultModule` you are supposed to use `createDefaultCoreModule`. A showcase repo is appreciated :)
